src/mdframe/reader.py

```diff
@@ -39,38 +42,50 @@ def data_dir_to_dataframes(data_dir_path: Path,
                            schema: Dict) -> List[pd.DataFrame]:
     metadata_file_paths = data_dir_path.glob(f"*.{metadata_file_extension}")    
     metadata_file_contents = [load_metadata_file(path, schema) for path in metadata_file_paths]
-    metadata_records = map(metadata_file_to_df, metadata_file_contents)
-    return list(metadata_records)
 
+    # remove all None(s) representing empty reads
+    metadata_file_contents = [text for text in metadata_file_contents if text != None]
+    
+    metadata_records = []
+    for text in metadata_file_contents:
+        metadata_records.append(pd.Series(text))
+
+    return metadata_records
+    
 @dataclass
 class Config:
     data_dir_path: Path
     metadata_file_extension: Optional[MetadataFileExtension]
     # data_file_extensions: List[DataFileExtension]
-    schema_url: Url | Path
+    schema_loc: Path | str
     __schema: Dict | None = None
 
     @property
     def schema(self):
         if self.__schema is None:
-            if isinstance(self.schema_url, Path):
-                if not self.schema_url.exists():
-                    raise ValueError(f"Path {self.schema_url} does not seem to point to a valid JSON schema file")
+            if isinstance(self.schema_loc, Path):
+                if not self.schema_loc.exists():
+                    raise ValueError(f"Path {self.schema_loc} does not seem to point to a valid JSON schema file")
 
-                with open(self.schema_url, "r") as f:
+                with open(self.schema_loc, "r") as f:
                     self.__schema = json.loads(f.read())
-            else: # schema_url is of type Url
+            else: # schema_url is a string (representing a URL)
```

Comment:
This is good but I'd use `elif` here and make sure that the string is an actual URL. You can use a URL validator, check this SO thread:
https://stackoverflow.com/questions/7160737/how-to-validate-a-url-in-python-malformed-or-not

Reply:
Done! I looked into validators and a lot of people were pointing out several issues. I figured I could just use urlparse from urllib3 since I already have urllib imported already.